next/app/webChatOptions.jsx

```diff
@@ -0,0 +1,321 @@
+
+const WebChatOptions = () => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+
+  const [selectedPrivacyOption, setSelectedPrivacyOption] = useState("disable");
+
+  const [selectedLauncherOption, setSelectedLauncherOption] =
+    useState("showLauncher");
+
+  const [position, setPosition] = useState("bottom-right");
+
+  const [selectedFullScreenOption, setSelectedFullScreenOption] =
+    useState("exitFullScreen");
+
+  const [unreadMessageCount, setMessageUnreadCount] = useState(0);
+
+  const [message, setMessage] = useState("");
+  const [messageEventListenerIsAdded, setmessageEventListenerIsAdded] =
+    useState(false);
+
+  const onLogin = () => {
+    // @NOTE: You can update the data in config object using the below code.
+    window.helpshiftConfig.userId = "captain_planet12";
+    window.helpshiftConfig.userEmail = "captain@example.com";
+
+    // @NOTE: If you want to update the helpshiftConfig object after your web page
+    //  has loaded, you can do it by calling the following API.
+    window.Helpshift("updateHelpshiftConfig");
+    setIsLoggedIn(true);
+    setMessage("");
+  };
+
+  const onLogout = () => {
+    window.helpshiftConfig.userId = "";
+    window.helpshiftConfig.userEmail = "";
+    window.Helpshift("updateHelpshiftConfig");
+    setIsLoggedIn(false);
+    setMessage("");
+  };
+
+  const onFullPrivacyChange = (e) => {
+    const val = e.target.value;
+    setSelectedPrivacyOption(val);
+    if (val === "enable") {
+      window.helpshiftConfig.fullPrivacy = true;
+    } else {
+      window.helpshiftConfig.fullPrivacy = false;
+    }
+    window.Helpshift("updateHelpshiftConfig");
+  };
+
+  const onLauncherChange = (e) => {
+    const val = e.target.value;
+    if (val == "showLauncher") {
+      // @NOTE: You can show Web Chat completely by calling the following API.
+      window.Helpshift("show");
+      setSelectedLauncherOption("showLauncher");
+    } else {
+      // @NOTE: You can hide Web Chat completely by calling the following API.
+      window.Helpshift("hide");
+      setSelectedLauncherOption("hideLauncher");
+    }
+  };
+
+  const onChangeWidgetPosition = (e) => {
+    const value = e.target.value;
+    setPosition(value);
+  };
+
+  const onWidgetPositionApply = () => {
+    window.helpshiftConfig.widgetOptions.position = position;
+    window.Helpshift("updateHelpshiftConfig");
+  };
+
+  const onFullScreenChange = (e) => {
+    const val = e.target.value;
+    setSelectedFullScreenOption(val);
+    if (val === "exitFullScreen") {
+      window.helpshiftConfig.widgetOptions.fullScreen = false;
+    } else {
+      window.helpshiftConfig.widgetOptions.fullScreen = true;
+    }
+
+    window.Helpshift("updateHelpshiftConfig");
+  };
+
+  // @NOTE: In order to add an event to the window, you should add it
+  // when the component mounts using the useEffect hook
+  // and remove it when the component unmounts.
+  useEffect(() => {
```

Comment:
@dv-Gogna 
Can you please add one more note here?

```
// @NOTE: Do not call window.Helpshift("updateHelpshiftConfig") inside useEffect. This may lead to unknown side effects.
```